nixops/backends/__init__.py

```diff
@@ -259,12 +260,26 @@ def wait_for_ssh(self, check=False):
         """Wait until the SSH port is open on this machine."""
         if self.ssh_pinged and (not check or self._ssh_pinged_this_time): return
         self.log_start("waiting for SSH...")
-        nixops.util.wait_for_tcp_port(self.get_ssh_name(), self.ssh_port, callback=lambda: self.log_continue("."))
-        self.log_end("")
-        if self.state != self.RESCUE:
-            self.state = self.UP
-        self.ssh_pinged = True
-        self._ssh_pinged_this_time = True
+
+        n = 0
+        timeout = 60
+
+        while True:
+            try:
+                self.ssh.run_command("exit", self.get_ssh_flags(),
+                    logged=False, timeout=timeout)
```

Comment:
yes. Also this changed introduced "global" timeout, after 60 tries it stops, when previous method was checking forever I think. But those are details to this change and I can adjust it as you prefer, for me the core idea here is to use SSH.